[ADD] eCommerce: add to cart page (#3175)
odoo/documentation: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst

===========
Add to cart
===========

There are multiple ways to add products to the **cart** and customize the actions of the

> **add-to-cart** button.

Author: Changed it up a bit

Process`, and select one of the three options under the :guilabel:`Add to Cart` section.

.. note::
   The action of the :guilabel:`Add to Cart` button is specific per website.

Hide the 'Add to Cart' button
=============================

If you do not need or wish to use the :guilabel:`Add to Cart` button, you can replace the button
with a :guilabel:`Contact Us` button which integrates a custom **redirect URL** of your choice.
When clicking on the :guilabel:`Contact Us` button, customers are taken to the specified **redirect
URL**.

To do so, go to :menuselection:`Configuration --> Settings --> Shop - Products` and tick
:guilabel:`Prevent Sale of Zero Priced Product`. This creates a new :guilabel:`Button url` field
where you can enter the **redirect URL** to be used. Then, set the price of the product to `0.00`
either from the **product's template**, or from a
:doc:`pricelist <../../../sales/sales/products_prices/prices/pricing>`.

.. tip::
   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
   purchases to customers with an account, but still want to display an online product catalog for

>    those without.

Author: Fair point :)

.. image:: cart/cart-contactus.png
   :align: center
   :alt: Contact us button on product page

Custom 'Add to Cart' button
===========================

You can also create a custom :guilabel:`Add to Cart` button and link it to a specific product.
The **custom button** can be added on any pages of the website as an **inner content** building
block, and is an *additional* button to the regular :guilabel:`Add to Cart` button.

To add it, go on the **product page** of your choice, click :menuselection:`Edit --> Blocks` and
place the building block. Once placed, you have the following options:

- :guilabel:`Product`: select the product to link the button with. Selecting a product renders the
  :guilabel:`Action` field available;
- :guilabel:`Action`: choose if the button should :guilabel:`Add to Cart` or :guilabel:`Buy Now`

>   (instant checkout).

Author: No changes in your suggestion, so as discussed I'm assuming it is a mistake :)

===========

> Add to cart

Author: I'm not sure I understand what you mean here. There's a new eCommerce structure, and it includes a checkout section :) : https://docs.google.com/spreadsheets/d/1O7Jg4T8kPE0HaLw-3YDjaj54MMk957-9akBFaSNKQA0/edit#gid=0

===========

The :guilabel:`Add to Cart` button can be customized in multiple ways to perform specific actions,
disabled to prevent sales, or redirect to a specific URL. Additionally, it can be combined with two
**extra buttons** such as the **"Buy Now" (instant checkout)** button and **"Re-order"** button to

> increase the number of 'buy' options for customers.

Author: Great suggestion

'Add to Cart' action customization
==================================

The :guilabel:`Add to Cart` button on the product page can be configured to also perform one the
following actions, on top of **adding to cart**:

- (Add to cart and) stay on the product page;
- (Add to cart and) redirect to the cart page;
- (Add to cart and) open a dialog box to let the user choose.

To configure the action button, go to :menuselection:`Configuration --> Settings --> Shop - Checkout

> Process`, and select one of the three options under the :guilabel:`Add to Cart` section.

Author: I like the way you did it better

.. note::
   The action of the :guilabel:`Add to Cart` button is specific per website.

> 

Author: Relevant comment, I removed the first note.

> Hide the 'Add to Cart' button

Author: Good suggestion

=============================

If you do not need or wish to use the :guilabel:`Add to Cart` button, you can replace it with a
:guilabel:`Contact Us` button which integrates a custom **redirect URL** of your choice. When
clicking on the :guilabel:`Contact Us` button, customers are taken to the specified **redirect

> URL**.

Author: True

To do so, go to :menuselection:`Configuration --> Settings --> Shop - Products` and tick
:guilabel:`Prevent Sale of Zero Priced Product`. This creates a new :guilabel:`Button url` field
where you can enter the **redirect URL** to be used. Then, set the price of the product to `0.00`
either from the **product's template**, or from a
:doc:`pricelist <../../../sales/sales/products_prices/prices/pricing>`.

.. note::
   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
   purchases only to customers with an account, but still want to display an online product catalog

>    for those without.

Author: It is intended for eCommerces that only sell B2B and not B2C, therefore actually choosing who can buy on their website by granting them portal access through a selection (from what I understood from the smartclass)

>    for those without.

Author: Same comment as above

>    for those without.

Author: See explanation above, and I added a todo:: for when the doc containing the mandatory sign up" is published

.. image:: cart/cart-contactus.png
   :align: center
   :alt: Contact us button on product page

> Custom 'Add to Cart' button

Author: I changed it to "Customizable" which defines better what the button really is about, however if you have any suggestions, lmk as I don't know how else to name a button can be modified by the user.

> Custom 'Add to Cart' button

Reviewer: You can customize default/standard add to cart buttons as well, so I felt talking about "additional" buttons might have been clearer

===========================

You can also create a custom :guilabel:`Add to Cart` button and link it to a specific product.
The **custom button** can be added on any page of the website as an **inner content** building
block, and is an *additional* button to the regular :guilabel:`Add to Cart` button.

> To add it, go on the **product page** of your choice, click :menuselection:`Edit --> Blocks` and

Reviewer: You can add it to any page.

Author: Reworded

place the building block. Once placed, you have the following options:

- :guilabel:`Product`: select the product to link the button with. Selecting a product renders the
  :guilabel:`Action` field available;
- :guilabel:`Action`: choose if the button should :guilabel:`Add to Cart` or :guilabel:`Buy Now`
  (instant checkout).

.. image:: cart/cart-add.png
   :align: center
   :alt: Custom 'Add to Cart' button

'Buy Now' button
================

> You can enable the :guilabel:`Buy Now` button to instantly take the customer to **checkout** instead

Reviewer: ```suggestion
You can enable the 'Buy Now' button to instantly take the customer to **checkout** instead
```

of adding the product to the cart. The :guilabel:`Buy Now` button is an *additional* button and
does not replace the :guilabel:`Add to Cart` button. To enable it, go to

> :menuselection:`Configuration --> Settings --> Shop - Checkout Process` and tick

Reviewer: ```suggestion
:menuselection:`Website --> Configuration --> Settings --> Shop - Checkout Process` and tick
```

:guilabel:`Buy Now`.

.. image:: cart/cart-buy-now.png
   :align: center
   :alt: Buy Now button

Re-order from portal
====================

Customers have the possibility to **re-order** items from **previous sales orders** on the customer

> portal. To do so, go to :menuselection:`Configuration --> Settings --> Shop - Checkout Process` and

Reviewer: ```suggestion
portal. To do so, go to :menuselection:`Website --> Configuration --> Settings --> Shop - Checkout Process` and
```

> enable :guilabel:`Re-order From Portal`. Customers can find the :guilabel:`Re-order` button on their

Reviewer: ```suggestion
enable :guilabel:`Re-order From Portal`. Customers can find the :guilabel:`Order Again` button on their
```

When customers click on the :guilabel:`Add to Cart` button, the product is added to their cart, and
customers remain **by default** on the product's page. However, customers can either immediately be
**redirected** to their cart, or given the choice on what to do through a **dialog box**.

To change the default behavior, go to :menuselection:`Website app --> Configuration --> Settings`.
Under the :guilabel:`Shop - Checkout Process` section, look for :guilabel:`Add to Cart` and select
one of the options.

.. note::
   If a product has **optional products**, the **dialog box** will always appear.

.. todo:: add :ref: to optional products when published.

Replace 'Add to Cart' button by 'Contact Us' button
===================================================

You can replace the 'Add to Cart' button with a 'Contact Us' button which redirects users to the URL
of your choice.

.. note::
   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
   purchases only to customers with an account, but still want to display an online product catalog
   for those without.

.. todo:: add :ref: to sign-up section doc when published to note

> To do so, go to :menuselection:`Website app --> Configuration --> Settings --> Shop - Products` and

Reviewer: ```suggestion
To do so, go to :menuselection:`Website --> Configuration --> Settings --> Shop - Products` and
```
It's not necessary to mention "app" - discussed that with Jon previously.
